src/com/facebook/buck/swift/SwiftDescriptions.java

```diff
@@ -69,4 +60,27 @@ private SwiftDescriptions() {
     output.enableObjcInterop = Optional.of(true);
   }
 
+  static boolean hasSwiftSource(ImmutableSortedSet<SourceWithFlags> srcs) {
+    return Iterables.any(srcs, new Predicate<SourceWithFlags>() {
+      @Override
+      public boolean apply(SourceWithFlags input) {
+        SourcePath sourcePath = input.getSourcePath();
+        return (sourcePath instanceof PathSourcePath) &&
```

Comment:
Just looked at the how `BuildTargetSourcePath` can be included in `srcs` (like the one in `genrule`), we don't know that path without a `BuildRuleResolver` (https://github.com/facebook/buck/blob/405d49e6685071f12884998c0b3066dcdcc67193/src/com/facebook/buck/rules/SourcePathResolver.java#L191-L191). But the case of using genrule to add swift source file is quite trivial at this moment, we can just ignore it for now. I'll revisit this again when working on project generation.